src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2728,6 +2728,13 @@ async def internal_datasets_filters(self):
                 ['id', '!=', sys_config['basename']],
                 ['id', '!^', f'{sys_config["basename"]}/'],
             ])
+
+        boot_pool = await self.middleware.call('boot.pool_name')
+        filters.extend([
+            ['id', '!=', boot_pool],
+            ['id', '!^', f'{boot_pool}/']
```

Comment:
If there isnt any performant way to filter these datasets, isnt it easier/better to use `pool != boot_pool`. I vaguely remember that being an attribute